src/emulator/cpu.ts

```diff
@@ -391,6 +417,74 @@ export class Cpu {
                 return instruction.cycles;
             }
 
+            case Operation.rlca: {
+                this.clock.increment(instruction.cycles);
+
+                const operand = this.state.r8[r8.a];
+                const result = ((operand << 1) | (operand >>> 7)) & 0xff;
+
+                this.state.r8[r8.a] = result;
+
+                // prettier-ignore
+                this.state.r8[r8.f] =
+                    (result === 0 ? flag.z : 0x00) |
```

Comment:
Sources differ here, but most sources that I can find seem to agree that Z is actually reset (which makes sort of sense to me, as the result cannot be zero if the source is not zero in the first place).